applications/portal/frontend/src/components/home/ExplorationSpinalCordDialog.tsx

```diff
@@ -36,35 +28,157 @@ const useStyles = makeStyles(() => ({
       marginBottom: '0.75rem'
     },
   },
+  editorWrapper: {
+    '&:hover': {
+      borderColor: '#fff'
+    },
+    '&:focus-within': {
+      borderColor: '#7b61ff'
+    },
+  },
 }));
 
+const NAME_KEY = "name";
+const DESCRIPTION_KEY = "description";
+
 export const ExplorationSpinalCordDialog = (props: any) => {
   const classes = useStyles();
-  const { open, handleClose } = props;
+  const commonClasses = common();
+  const api = WorkspaceService.getApi();
+  const { experimentId, experiment, tagsOptions, open, handleClose, refreshExperimentList, setExperimentMenuEl } = props;
+
+  const [name, setName] = React.useState<string>(experiment.name);
+  const [description, setDescription] = React.useState<string>(experiment.description);
+  const [validationErrors, setValidationErrors] = React.useState(new Set([]));
+  const [initialTags, setInitialTags] = React.useState(experiment.tags.map((t: any) => t.name));
```

Comment:
We don't need the initialTags state variable. You can, and it's probably better if you do (since it avoids consistency problems), just use the experiment.tags. 
After the experiment update, the new experiments fetch that happens will contain the experiment latest state with the latest tags - assuming the delete tags is working which doesn't seem like it is but I will investigate that myself